common/core/gpudevice.cpp

```diff
@@ -202,6 +202,11 @@ void GpuDevice::DisplayManager::HotPlugEventHandler() {
       UpdateDisplayState();
     }
   }
+  /* Adding sleep to avoid busy loop. The busy loop returns only
+   * if read() completes / fails. Race conditions can cause this
+   * busy loop hog the system. This 1 ms sleep can avoid that.
+   */
+  usleep(1000);
```

Comment:
I don't think we should usleep here. In case this is really required then we better look what is causing the busyloop and fix that.